api/src/repositories/base-repository.ts

```diff
@@ -12,4 +14,12 @@ export class BaseRepository {
   constructor(connection: IDBConnection) {
     this.connection = connection;
   }
+
+  async isSystemUserAdmin(): Promise<boolean> {
+    const userService = new UserService(this.connection);
+    const userObject = await userService.getUserById(this.connection.systemUserId());
+    return [SYSTEM_ROLE.SYSTEM_ADMIN, SYSTEM_ROLE.DATA_ADMINISTRATOR].some((systemRole) =>
+      userObject.role_names.includes(systemRole)
+    );
```

Comment:
Moving this to a reusable function is a good call.

It does spark some thoughts though about my original setup with the search repository functions. I don't think I should have done the admin check at the repository level, I should have done that at the service level. I'm thinking the repositories should probably never call the services, as they should be kept simple and focused only on building/running SQL queries.

If you're cool with refactoring this a bit, I think it would be best to move this `isSystemUserAdmin` function into the user service, and then move the admin checks we are currently doing in the spatial repository up to the spatial service. And then there will be no reference to any services in the spatial-repository, and the functions that do the admin checks currently can be removed.

Reply:
Good call moving it out of the repos. That logic lives in the services now. Tests are also updated to reflect those changes

Reply:
Just looked through it all, looks really good!